test/unit/name.js

```diff
@@ -0,0 +1,158 @@
+    });
+    this._getMACAddress = sinon.stub(Tessel.prototype, "_getMACAddress", function(callback) {
+      callback(null, "TheFakeName");
+    });
+
+    this.isValidName = sinon.spy(Tessel, "isValidName");
+    this.renameTessel = sinon.spy(controller, "renameTessel");
+    this.setName = sinon.spy(Tessel.prototype, "setName");
+    this.setHostname = sinon.spy(commands, "setHostname");
+    this.getHostname = sinon.spy(commands, "getHostname");
+    this.logsWarn = sinon.stub(logs, "warn", function() {});
+    this.logsInfo = sinon.stub(logs, "info", function() {});
+
+    this.tessel = new Tessel();
+    this.tessel.connection = {
+      exec: sinon.spy()
+    };
+
+    done();
+  },
+
+  tearDown: function(done) {
+    this.isValidName.restore();
+    this.renameTessel.restore();
+    this.getName.restore();
+    this.setName.restore();
+    this._getMACAddress.restore();
+    this.setHostname.restore();
+    this.getHostname.restore();
+    this.logsWarn.restore();
+    this.logsInfo.restore();
+    done();
+  },
+
+  isValidName: function(test) {
+    test.expect(2);
+
+    // This needs more fleshing out.
+    //
+    test.equal(Tessel.isValidName("foo"), true);
+    test.equal(Tessel.isValidName("foo-"), false);
+
+    test.done();
+  },
+
+  renameTesselNoOpts: function(test) {
+    test.expect(1);
+
+    this.renameTessel().catch(function(error) {
+      test.equal(error, "A new name must be provided.");
+      test.done();
+    });
+  },
+
+  renameTesselInvalid: function(test) {
+    test.expect(1);
+
+    this.renameTessel({ newName: "!@#$" }).catch(function(error) {
+      test.equal(error, "Invalid name: !@#$. The name must be a valid hostname string. See http://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names.");
+      test.done();
+    });
+  },
+
+  resetName: function(test) {
+    test.expect(6);
+
+    this.tessel.rename({ reset: true });
+
+    // When reset:
+    // - the mac address is requested
+    // - setName is called
+    // - the connection executes the setHostName command
+    test.equal(this._getMACAddress.callCount, 1);
+    test.equal(this.setName.callCount, 1);
+    test.equal(this.tessel.connection.exec.callCount, 1);
+    test.equal(this.setHostname.callCount, 1);
+    test.ok(this.setHostname.lastCall.calledWith("Tessel-TheFakeName"));
+
+    // getName is _not_ called.
+    test.equal(this.getName.callCount, 0);
+
+    test.done();
+  },
+
+  validRename: function(test) {
+    // test.expect(3);
+
+    this.tessel.rename({ newName: "ValidAndUnique" });
+
+    // When valid rename:
+    // - getName is called
+    // - setName is called
+    // - the connection executes the setHostName command
+    test.equal(this.getName.callCount, 1);
+    test.equal(this.setName.callCount, 1);
+    test.equal(this.tessel.connection.exec.callCount, 1);
+    test.equal(this.setHostname.callCount, 1);
+    test.ok(this.setHostname.lastCall.calledWith("ValidAndUnique"));
```

Comment:
It won't have actually done that, since that step is stubbed. This is just testing that renaming is validated and valid names pass